cc/algorithms/bounded-mean.h

```diff
@@ -209,7 +209,7 @@ class BoundedMean : public Algorithm<T> {
     return memory;
   }
 
- private:
+ protected:
```

Comment:
I think you only care about making the constructor protected - you should keep the rest of the private members and functions private.

Reply:
Don't we want the algorithms to be consistent? The GenerateResult and ResetState functions are also protected (not private) in Bounded Sum.

Reply:
Update: I made all the mechanisms-related functions and variables private but kept the above functions protected.

Reply:
That sounds like the right approach - I think you're right to make GenerateResult and ResetState protected for consistency, but the member variables need to be private per the style guide (and BuildSumMechanism doesn't seem likely to be useful to a subclass).